FabFos.py

```diff
@@ -266,7 +279,9 @@ def get_options():
                       help="Increase the level of verbosity in runtime log.")
     opts.add_argument("-h", "--help",
                       action="help", help="Show this help message and exit")
-
+    if args.type == "B":
+        args.reads = bam2fastq(args.reads, sample.id, sample.output_dir, args.executables)
+    
```

Comment:
This code should come outside of the `get_options` function. I suggest immediately after the call to `get_options` in the main function of FabFos.